### PR TITLE
Rewrite Luigi's navigation footer

### PR DESCRIPTION
**Pull request by Lyczeq**
**Description**

Changes proposed in this pull request:

- Added a Footer component with the same logic as it is in Luigi
- Added a feature flag to switch between Luigi and React navigation

**Related issue(s)**
See #1939

### CODE REVIEW

File: core-ui/src/sidebar/Footer/useGetBusolaVersion.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect } from 'react';
+import { useTranslation, TFunction } from 'react-i18next';
+import jsyaml from 'js-yaml';
+
+const BUSOLA_GITHUB_REPOSITORY_LINK = 'https://github.com/kyma-project/busola';
+const BUSOLA_GITHUB_PULLS_LINK = 'https://github.com/kyma-project/busola';
+const BUSOLA_GITHUB_COMMMITS_LINK = 'https://github.com/kyma-project/busola';
```
Comment (wchrapka) [inline core-ui/src/sidebar/Footer/useGetBusolaVersion.ts:7]: triple MMM

File: core-ui/src/sidebar/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { useGetLegalLinks } from './useGetLegalLinks';
+
+export function Footer() {
+  const { t } = useGetTranslation();
+  const { githubLink, busolaVersion } = useGetBusolaVersionDetails();
+  const legalLinks = useGetLegalLinks();
+
+  return (
+    <footer>
+      <div>
```
Comment (wchrapka) [inline core-ui/src/sidebar/Footer/Footer.tsx:13]: this div is probably not needed

File: core-ui/src/sidebar/Footer/useGetLegalLinks.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { useMicrofrontendContext } from 'shared/contexts/MicrofrontendContext';
+
+type LegalLink = {
+  label: string;
+  link: string;
+};
+
+export const useGetLegalLinks = (): LegalLink[] => {
+  const { t, i18n } = useTranslation();
+  const { features } = useMicrofrontendContext() as any;
```
Comment (wchrapka) [inline core-ui/src/sidebar/Footer/useGetLegalLinks.ts:11]: So, as the microfrontendContext is going to be dropped, the new components should subscribe to the recoil store. There is already a recoil atom for features `configFeaturesAtom` . Actually maybe I'll rewrite the features state today

File: core-ui/src/sidebar/Sidebar.js [hunk 4]
```diff
@@ -2,11 +2,14 @@ import React from 'react';
...
 import { useMicrofrontendContext } from 'shared/contexts/MicrofrontendContext';
 import { ErrorBoundary } from 'shared/components/ErrorBoundary/ErrorBoundary';
+import { Footer } from './Footer/Footer';
 
 export const Sidebar = () => {
-  const { kubeconfig } = useMicrofrontendContext();
+  const { features } = useMicrofrontendContext();
+
+  // turn on the feature locally to work on the new Navigation
+  if (!features?.REACT_NAVIGATION?.isEnabled) return null;
```
Comment (wchrapka) [inline core-ui/src/sidebar/Sidebar.js:11]: this is temporary so this is okay.

File: core-ui/public/assets/version.yaml [hunk 5]
```diff
@@ -0,0 +1 @@
+version: dev
```
Comment (wchrapka) [inline core-ui/public/assets/version.yaml:1]: what does that do? I think this works like the file is mounted by a config map on prod? But probably not yet. You can take the info if this is development or not without a requst by checking the global `process.env.NODE_ENV`. 

I'm not sure about this approach but talked with Piotr and agreed that as long as we are mounting the files in Luigi, we should take them from Luigi? Maybe this mount can be already moved to React's assets

File: core-ui/src/sidebar/Footer/useGetBusolaVersion.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      const busolaVersion = await getBusolaVersion(t);
+      const githubLink = createGithubLink(busolaVersion);
+
+      setVersionDetails({
+        busolaVersion,
+        githubLink,
+      });
+    };
+
+    getVersion();
```
Comment (wchrapka) [inline core-ui/src/sidebar/Footer/useGetBusolaVersion.ts:58]:
```suggestion
    void getVersion();
```

File: core-ui/src/sidebar/Footer/useGetBusolaVersion.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (wchrapka) [inline core-ui/src/sidebar/Footer/useGetBusolaVersion.ts:59]: `t` has a constant identity so you don't need to disable eslint here.

File: resources/web/deployment.yaml [hunk 8]
```diff
@@ -16,7 +16,7 @@ spec:
     spec:
       containers:
         - name: busola
-          image: eu.gcr.io/kyma-project/busola-web:PR-1974
+          image: eu.gcr.io/kyma-project/busola-web:PR-1969
```
Comment (wchrapka) [inline resources/web/deployment.yaml:19]: It's a not a big task, why not style it all together?

**Review by wchrapka — CHANGES_REQUESTED** [review]

File: core-ui/public/assets/version.yaml (hunk 5 above)
Comment (Lyczeq) [reply core-ui/public/assets/version.yaml:1]: This is a temporary solution until we remove Luigi. More info [here](https://github.com/kyma-project/busola/issues/1924#issuecomment-1286885130).

**Review by wchrapka — APPROVED** [review]

**Pull request merged**